irohad/main/application.cpp

```diff
@@ -52,24 +55,33 @@ Irohad::Irohad(const std::string &block_store_dir,
       keypair(keypair) {
   log_ = logger::log("IROHAD");
   log_->info("created");
+  // Initializing storage at this point in order to read genesis block before
+  // initialization of iroha deamon
```

Comment:
It is because all of the init* functions are placed in Irohad::init() except for initStorage. It seems a bit non consistent, but there is a reason for it. I tried to specify the reason.